### ISSUE TITLE
Redundant output

### ISSUE THREAD

**Issue by migmartri** [vmware-tanzu/asset-relocation-tool-for-kubernetes#32]
Hi,

The command output, in my opinion, contains redundant information. For example, the following terminal output tells you twice the images that are going to be pushed.

```
Checking projects.registry.vmware.com/relocated/example1/mariadb:10.5.12-debian-10-r0 (sha256:8e2c533c786b7c921c75a4f26f1779362bdbb3a38ffd0e7771a93078eb641692)...
Push required
Checking projects.registry.vmware.com/relocated/example1/mysqld-exporter:0.13.0-debian-10-r56 (sha256:a591b9f9c08b328efd3bd5815c275c348e1a631688aec984e32185946d84126f)...
Push required
Checking projects.registry.vmware.com/relocated/example1/bitnami-shell:10-debian-10-r153 (sha256:53891aea23bdc9fe2d1aa248ea260c867b51eb07e26debc01c00fb7169208950)...
Push required

Images to be pushed:
  projects.registry.vmware.com/relocated/example1/mariadb:10.5.12-debian-10-r0 (sha256:8e2c533c786b7c921c75a4f26f1779362bdbb3a38ffd0e7771a93078eb641692)
  projects.registry.vmware.com/relocated/example1/mysqld-exporter:0.13.0-debian-10-r56 (sha256:a591b9f9c08b328efd3bd5815c275c348e1a631688aec984e32185946d84126f)
  projects.registry.vmware.com/relocated/example1/bitnami-shell:10-debian-10-r153 (sha256:53891aea23bdc9fe2d1aa248ea260c867b51eb07e26debc01c00fb7169208950)
```

To me, the first part of the output "checking..." might be more suitable to a debug mode via a debug `verbosity` flag, while the second part is indeed part of the stdout that we want the user to read.

I'd suggest to remove the first part of the output from the default stdout and extend the second part with images that do not need to be pushed. In fact I do not see any value indicating what images are going to be pushed and which ones not. I'll be ok just indicating the destination.

In other words, I'd just show something like

```
origin image 1 =>  destination image 1
origin image 2 => destination image 2
...
```

That way you can see at first glance (as part of the information shown just before the prompt) the images that the system has resolved and the final destination of these.

I understand that showing "pulling/checking x" indicates progress but I'd argue that they are not that important, since that's not what usually takes longer, maybe the pulling images can be heavy, but we might be able to replace it by an API check to get the digest, which is what we need. I'd remove all those lines and just showing "found x images, calculating rewrites..."

WDYT?

cc/ @petewall @josvazg

**Comment by josvazg** [comment]
Makes sense, I am taking this one.

**Comment by josvazg** [comment]
Sample output BEFORE any changes:
```
$ ./build/relok8s chart move test/fixtures/wordpress-11.0.4.tgz -i test/fixtures/wordpress-11.0.4.images.yaml --registry harbor-repo.vmware.com --repo-prefix josvaz
Pulling index.docker.io/bitnami/wordpress:5.7.2-debian-10-r0...
Done
Pulling index.docker.io/bitnami/apache-exporter:0.8.0-debian-10-r378...
Done
Pulling index.docker.io/bitnami/bitnami-shell:10...
Done
Pulling index.docker.io/bitnami/mariadb:10.5.10-debian-10-r0...
Done
Pulling index.docker.io/bitnami/mysqld-exporter:0.12.1-debian-10-r430...
Done
Pulling index.docker.io/bitnami/memcached:1.6.9-debian-10-r140...
Done
Pulling index.docker.io/bitnami/memcached-exporter:0.9.0-debian-10-r26...
Done
Checking harbor-repo.vmware.com/josvaz/wordpress:5.7.2-debian-10-r0 (sha256:0a8168532ef9ef4cd99270913c253c7bfcc7f98fd3377165ad491baef63d7c3c)...
Push required
Checking harbor-repo.vmware.com/josvaz/apache-exporter:0.8.0-debian-10-r378 (sha256:018141faec368d95df77b1c6bf6ad7aff82dcebf809b1b675bc7edfa7def0dfb)...
Push required
Checking harbor-repo.vmware.com/josvaz/bitnami-shell:10 (sha256:26cea5bce5e17be7007ee4932ecbb3d5cb4514d715169da38d8f35c3f981e52b)...
Push required
Checking harbor-repo.vmware.com/josvaz/mariadb:10.5.10-debian-10-r0 (sha256:0244c6b04481971fe1798e3faf25382e32273c6856d01fdd55cc42a3d58d4ec3)...
Push required
Checking harbor-repo.vmware.com/josvaz/mysqld-exporter:0.12.1-debian-10-r430 (sha256:b8a3ce87c578c5d235e584ef51a67dc02d4395fae90b2fca613d6fac6d4d2d32)...
Push required
Checking harbor-repo.vmware.com/josvaz/memcached:1.6.9-debian-10-r140 (sha256:ad06e83e5fb0f31640db9e838d7d1bab03575b4701b3c7600f5a77a456655d4b)...
Push required
Checking harbor-repo.vmware.com/josvaz/memcached-exporter:0.9.0-debian-10-r26 (sha256:3eea60049054c04fe4154044900a98652cbb607746f56b1a8e29664da05da1a6)...
Push required

Images to be pushed:
  harbor-repo.vmware.com/josvaz/wordpress:5.7.2-debian-10-r0 (sha256:0a8168532ef9ef4cd99270913c253c7bfcc7f98fd3377165ad491baef63d7c3c)
  harbor-repo.vmware.com/josvaz/apache-exporter:0.8.0-debian-10-r378 (sha256:018141faec368d95df77b1c6bf6ad7aff82dcebf809b1b675bc7edfa7def0dfb)
  harbor-repo.vmware.com/josvaz/bitnami-shell:10 (sha256:26cea5bce5e17be7007ee4932ecbb3d5cb4514d715169da38d8f35c3f981e52b)
  harbor-repo.vmware.com/josvaz/mariadb:10.5.10-debian-10-r0 (sha256:0244c6b04481971fe1798e3faf25382e32273c6856d01fdd55cc42a3d58d4ec3)
  harbor-repo.vmware.com/josvaz/mysqld-exporter:0.12.1-debian-10-r430 (sha256:b8a3ce87c578c5d235e584ef51a67dc02d4395fae90b2fca613d6fac6d4d2d32)
  harbor-repo.vmware.com/josvaz/memcached:1.6.9-debian-10-r140 (sha256:ad06e83e5fb0f31640db9e838d7d1bab03575b4701b3c7600f5a77a456655d4b)
  harbor-repo.vmware.com/josvaz/memcached-exporter:0.9.0-debian-10-r26 (sha256:3eea60049054c04fe4154044900a98652cbb607746f56b1a8e29664da05da1a6)

Changes written to wordpress/values.yaml:
  .image.registry: harbor-repo.vmware.com
  .image.repository: josvaz/wordpress
  .metrics.image.registry: harbor-repo.vmware.com
  .metrics.image.repository: josvaz/apache-exporter
  .volumePermissions.image.registry: harbor-repo.vmware.com
  .volumePermissions.image.repository: josvaz/bitnami-shell

Changes written to wordpress/charts/mariadb/values.yaml:
  .mariadb.image.registry: harbor-repo.vmware.com
  .mariadb.image.repository: josvaz/mariadb
  .mariadb.metrics.image.registry: harbor-repo.vmware.com
  .mariadb.metrics.image.repository: josvaz/mysqld-exporter
  .mariadb.volumePermissions.image.registry: harbor-repo.vmware.com
  .mariadb.volumePermissions.image.repository: josvaz/bitnami-shell

Changes written to wordpress/charts/memcached/values.yaml:
  .memcached.image.registry: harbor-repo.vmware.com
  .memcached.image.repository: josvaz/memcached
  .memcached.metrics.image.registry: harbor-repo.vmware.com
  .memcached.metrics.image.repository: josvaz/memcached-exporter
Would you like to proceed? (y/N)
n
Aborting
```

**Comment by josvazg** [comment]
Sample output from prototype code:
```
$ ./build/relok8s chart move test/fixtures/wordpress-11.0.4.tgz -i test/fixtures/wordpress-11.0.4.images.yaml --registry harbor-repo.vmware.com --repo-prefix josvaz
Computing relocation...

Image moves:
 index.docker.io/bitnami/wordpress:5.7.2-debian-10-r0 => harbor-repo.vmware.com/josvaz/wordpress:5.7.2-debian-10-r0 (sha256:0a8168532ef9ef4cd99270913c253c7bfcc7f98fd3377165ad491baef63d7c3c) (push required)
 index.docker.io/bitnami/apache-exporter:0.8.0-debian-10-r378 => harbor-repo.vmware.com/josvaz/apache-exporter:0.8.0-debian-10-r378 (sha256:018141faec368d95df77b1c6bf6ad7aff82dcebf809b1b675bc7edfa7def0dfb) (push required)
 index.docker.io/bitnami/bitnami-shell:10 => harbor-repo.vmware.com/josvaz/bitnami-shell:10 (sha256:26cea5bce5e17be7007ee4932ecbb3d5cb4514d715169da38d8f35c3f981e52b) (push required)
 index.docker.io/bitnami/mariadb:10.5.10-debian-10-r0 => harbor-repo.vmware.com/josvaz/mariadb:10.5.10-debian-10-r0 (sha256:0244c6b04481971fe1798e3faf25382e32273c6856d01fdd55cc42a3d58d4ec3) (push required)
 index.docker.io/bitnami/mysqld-exporter:0.12.1-debian-10-r430 => harbor-repo.vmware.com/josvaz/mysqld-exporter:0.12.1-debian-10-r430 (sha256:b8a3ce87c578c5d235e584ef51a67dc02d4395fae90b2fca613d6fac6d4d2d32) (push required)
 index.docker.io/bitnami/bitnami-shell:10 => harbor-repo.vmware.com/josvaz/bitnami-shell:10 (sha256:26cea5bce5e17be7007ee4932ecbb3d5cb4514d715169da38d8f35c3f981e52b)
 index.docker.io/bitnami/memcached:1.6.9-debian-10-r140 => harbor-repo.vmware.com/josvaz/memcached:1.6.9-debian-10-r140 (sha256:ad06e83e5fb0f31640db9e838d7d1bab03575b4701b3c7600f5a77a456655d4b) (push required)
 index.docker.io/bitnami/memcached-exporter:0.9.0-debian-10-r26 => harbor-repo.vmware.com/josvaz/memcached-exporter:0.9.0-debian-10-r26 (sha256:3eea60049054c04fe4154044900a98652cbb607746f56b1a8e29664da05da1a6) (push required)

Changes written to wordpress/values.yaml:
  .image.registry: harbor-repo.vmware.com
  .image.repository: josvaz/wordpress
  .metrics.image.registry: harbor-repo.vmware.com
  .metrics.image.repository: josvaz/apache-exporter
  .volumePermissions.image.registry: harbor-repo.vmware.com
  .volumePermissions.image.repository: josvaz/bitnami-shell

Changes written to wordpress/charts/mariadb/values.yaml:
  .mariadb.image.registry: harbor-repo.vmware.com
  .mariadb.image.repository: josvaz/mariadb
  .mariadb.metrics.image.registry: harbor-repo.vmware.com
  .mariadb.metrics.image.repository: josvaz/mysqld-exporter
  .mariadb.volumePermissions.image.registry: harbor-repo.vmware.com
  .mariadb.volumePermissions.image.repository: josvaz/bitnami-shell

Changes written to wordpress/charts/memcached/values.yaml:
  .memcached.image.registry: harbor-repo.vmware.com
  .memcached.image.repository: josvaz/memcached
  .memcached.metrics.image.registry: harbor-repo.vmware.com
  .memcached.metrics.image.repository: josvaz/memcached-exporter
Would you like to proceed? (y/N)
```

**Comment by josvazg** [comment]
If we accept that change we may want to decide if we want to make a simplification on our API, before we actually go GA here and we cannot make these kind of backward-incompatible changes to consumers.

Right now we have:
```
func NewChartMover(chart *chart.Chart, patterns string, rules *rewrite.Rules, log Logger) (*ChartMover, error)

func (cm *ChartMover) Move(toChartFilename string) error

func (cm *ChartMover) Print()
```

We are passing `log Logger` to `NewChartMover` but after this change it is not used any more there. It is still required as it is recorded in the `ChartMover` structs to be used in the `Print()` and `Move()` methods.

Maybe we could also remove output from the move part, and leave it only to the Print. So the interface could be instead:
```
func NewChartMover(chart *chart.Chart, patterns string, rules *rewrite.Rules) (*ChartMover, error)

func (cm *ChartMover) Move(toChartFilename string) error

func (cm *ChartMover) Print(log Logger)
```

Basically we do not need to pass or record the logger, we only pass it to print as it will be the only one using it. And it makes more sense to pass a logger to a printer method.

WDYT @migmartri  & @petewall ?

**Comment by migmartri** [comment]
I am currently working on a proposal for some changes, I'll have a PR today that we can discuss over. But one of the things it does is to re-evaluate the use of the logger interface.

**Comment by josvazg** [comment]
Need to fix the external tests.
